x-pack/auditbeat/processors/add_session_metadata/config.go

```diff
@@ -0,0 +1,22 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+
+package add_session_metadata
+
+// Config for add_session_metadata processor.
+type Config struct {
+	Backend       string `config:"backend"`
+	ReplaceFields bool   `config:"replace_fields"`
+	PidField      string `config:"pid_field"`
+}
+
+func defaultConfig() Config {
+	return Config{
+		Backend:       "ebpf",
```

Comment:
If in the future we want to have the backend be automatic based on the kernel then we might want to allow two options now -- `auto` and `ebfp` with `auto` as the default. This way there doesn't need to be a breaking change later to enable the automatic backend behavior by default.

Reply:
I've set "auto" as the backend. It currently always uses ebpf, without any checks if ebpf is supported. In the future when more backends are added, I'll update auto to check for support